cfg_mgmt/gcp.py

```diff
@@ -53,10 +54,19 @@ def delete_service_account_key(
 
 
 def rotate_cfg_element(
-    cfg_element: model.container_registry.ContainerRegistryConfig,
+    cfg_element: model.container_registry.ContainerRegistryConfig | model.gcp.GcpServiceAccount,
     cfg_factory: model.ConfigFactory,
 ) ->  typing.Tuple[cfg_mgmt.revert_function, dict, model.NamedModelElement]:
-    client_email = json.loads(cfg_element.password())['client_email']
+    if not type(cfg_element) in (
+        model.container_registry.ContainerRegistryConfig,
+        model.gcp.GcpServiceAccount,
+    ):
+        raise TypeError(cfg_element)
+
+    if isinstance(cfg_element, model.container_registry.ContainerRegistryConfig):
+        client_email = json.loads(cfg_element.password())['client_email']
+    elif isinstance(cfg_element, model.gcp.GcpServiceAccount):
+        client_email = cfg_element.service_account_key()['client_email']
```

Comment:
rather add `else: raise ValueError` here (if you do, you might drop the initial value-checking at the beginning of the function.
rationale: there are multiple places where additional, typespecific handling is required, which all need to be considered. should a third option be added, there is a change extension of typespecific handling might be missed, resulting, potentially in a non-obvious follow-up error later